modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/security/DoubleSignDetect.scala

```diff
@@ -0,0 +1,12 @@
+package org.tessellation.node.shared.domain.security
+
+import org.tessellation.node.shared.domain.fork.ForkInfoEntries
+import org.tessellation.schema.peer.PeerId
+
+trait DoubleSignDetect[F[_]] {
```

Comment:
If it is a `service`, then it's fine to have a trait, but I'd move it from `domain/security` to something else, cluster or snapshot related, and to `service` directory. If it is a `program`, then there is no need to write a trait.

Reply:
Per our offline discussion, it is a Program. The trait has been removed, `DoubleSignDetect`'s implementation has been remade into a class and moved out of `infrastructure` and into `domain`.